File: csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n) =>
        n <= 0 || n > 64
            ? throw new ArgumentOutOfRangeException(nameof(n), n, "The numbers of square should be between 1 and 64.")
            : (ulong)Math.Pow(2, n - 1);

    public static ulong Total() => unchecked((ulong) Math.Pow(2, 64) - 1);
}

File: javascript/grains/grains.js
/**
 * You can use the bigint type and BigInt global object to support numbers below
 * Number.MIN_SAFE_INTEGER and above NUMBER.MAX_SAFE_INTEGER.
 *
 * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
 */

//
// This is only a SKELETON file for the 'Grains' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
export const square = (number) => {
  if (number <= 0 || number > 64) {
    throw new Error("square must be between 1 and 64");
  }

  return 1n << BigInt(number - 1);
};

export const total = () => 2n ** 64n - 1n;
